js/ui/windowManager.js

```diff
@@ -571,6 +586,44 @@ const TilePreview = new Lang.Class({
     }
 });
 
+const EnableBackfaceCullingEffect = new Lang.Class({
+    Name: 'EnableBackfaceCullingEffect',
+    Extends: Clutter.Effect,
+    _init : function(params) {
+        this.parent(params);
+    },
+    vfunc_pre_paint: function() {
+        this.get_actor().set_cull_back_face(true);
+    },
+    vfunc_post_paint: function() {
+        this.get_actor().set_cull_back_face(true);
+    }
+});
+
+/**
+ * Determine all windows belonging to a certain process id
+ */
+function windowActorsFromPid(pid) {
+    return global.get_window_actors().filter(function(window_actor) {
+        return window_actor.get_meta_window().get_pid() == pid;
+    });
+}
+
+/**
+ * For a given process id, determine the corresponding window
+ * (if any) and its size.
+ */
+function pidToActorInfo(pid) {
+    const windowActors = windowActorsFromPid(pid);
+
+    return {
+        actor: windowActors.length ? windowActors[0] : null,
```

Comment:
If I understand the code correctly, what you call here `actor`, you later refer to as `window`. So the code will effectively only end up working when none of the two apps are started.